### DISCUSSION TITLE
Проблема аддона на 19.4 Kodi

### DISCUSSION THREAD

**Discussion by ktchatskiy** [quarckster/kodi.kino.pub#232] [Q&A]
Аддон был установлен на LIbreELEC и все прекрасно работало, заметил проблемы недели с 2 назад. При попытке попасть в любое меню аддона ничего не происходит. Снес аддон и поставил заново. Теперь при попытке авторизовать устройство тоже нет никаких изменений. Ошибки ниже:

**Comment by stavultras** [comment]
Судя по логам. это не проблема аддона, а вашей сети.

![image](https://user-images.githubusercontent.com/6468524/199968400-644e9b5c-466e-473b-833b-1655993a3b0b.png)
Как минимум, это не имеет никакого отношения к аддону.

**Reply by ktchatskiy** [reply]
> Судя по логам. это не проблема аддона, а вашей сети.

с ПК все ок... именно LibreElEC выдает такие чудеса.
Лог приложил

**Reply by stavultras** [reply]
Судя по этому куску:
2022-11-04 15:02:02.319 T:1072     INFO <general>: [video.kino.pub]: sending payload {'grant_type': 'device_code', 'client_id': 'xbmc', 'client_secret': 'cgg3gtifu46urtfp2zp1nqtba0k2ezxh'} to oauth api
2022-11-04 15:02:05.167 T:1041    ERROR <general>: CCurlFile::Open failed with code 404 for http://feeds.kodi.tv/latest_xbmc_addons:

Аддон лезет вообще не туда, куда надо. Что-то перехватает URL и всё время кидает на http://feeds.kodi.tv/latest_xbmc_addons
Ищите проблему у себя

P.S.: У меня без проблем работает на Coreelec и Kodi 19.4.

**Reply by ktchatskiy** [reply]
> Аддон лезет вообще не туда, куда надо. Что-то перехватает URL и всё время кидает на http://feeds.kodi.tv/latest_xbmc_addons
> Ищите проблему у себя

это я видел, но если бы мог смотреть, тогда бы решил что и как) А так и прошу помощи....

**Reply by stavultras** [reply]
ну я так поднимаю, главная проблема: Error Contents: <urlopen error [Errno 110] Connection timed out>
Предудущие ошибки, то видимо сам Libreelec куда-то лезет и просто так в логах совпало, что оно подряд идёт.

Проверьте, работает ли у вас https://api.service-kp.com/v1 через wget в самом Libreelec через SSH.

**Reply by stavultras** [reply]
Должно быть что-то в духе:
wget https://api.service-kp.com/v1
```
--2022-11-04 15:11:10--  https://api.service-kp.com/v1
Resolving api.service-kp.com (api.service-kp.com)... 5.39.48.32
Connecting to api.service-kp.com (api.service-kp.com)|5.39.48.32|:443... connected.
HTTP request sent, awaiting response... 404 Not Found
2022-11-04 15:11:11 ERROR 404: Not Found.
```
Вместо timeout. 404 тут это абсолютно нормально. так как не переданы никакие данные.

**Reply by ktchatskiy** [reply]
> Проверьте, работает ли у вас https://api.service-kp.com/v1 через wget в самом Libreelec через SSH.

проверяю... судя по тому что еще думает, то проблема есть.
~ # wget https://api.service-kp.com/v1
Connecting to api.service-kp.com (5.39.48.32:443)
wget: error getting response: Connection reset by peer
:~ #

**Reply by ktchatskiy** [reply]
Черт побери, с провайдером ничего не меняли, с инетом все ок... с ПК кинопаб работает. Что могло измениться ума не приложу...

**Reply by stavultras** [reply]
а с ПК всё работает? может забанено провайдером, а на ПК вы пользуетесь VPN? Я просто других объяснений не особо вижу, так как приставка доступ в инет имеет. судя по:
```
CCurlFile::Open failed with code 404 for http://feeds.kodi.tv/latest_xbmc_addons
```
Тут 404, а не timeout.

**Reply by ktchatskiy** [reply]
> а с ПК всё работает?

использую без ВПН но с зеркалом личным. Как проверить без зеркала доступность?

**Reply by stavultras** [reply]
вы на ПК в браузере пользуетесь? или на ПК + Kodi? Браузер и Kinopub API - 2 разные вещи и для API нет зеркал, у него всегда один адрес. У меня на Ростелекоме он переодически попадает в бан, видимо по IP, ибо через какое-то время разбанивается.

**Reply by ktchatskiy** [reply]
> вы на ПК в браузере пользуетесь? или на ПК + Kodi?

Сам КиноПаб через личную ссылку открываю без VPN с ПК в браузере на ОС Win10. А логи и проблема с устройства с LibreELEC 10.0.3, на котором тоже все работало недели 2 назад, а потом вот такие приключения.

**Reply by stavultras** [reply]
Ну вот и объяснение, на компе работает из-за зеркала самого сайта, а в libreelec не работает, потому что ваш провайдер забанил Kinopub API URL, по всей видимости. Ибо на Libreelec к другим доменам доступ есть (хотя бы судя по ошибкам 404 от других URL)

**Reply by stavultras** [reply]
А попробуйте на ПК открыть https://api.service-kp.com/v1
Думаю, что не откроется.

**Reply by ktchatskiy** [reply]
> А попробуйте на ПК открыть https://api.service-kp.com/v1
> Думаю, что не откроется.

ERR_CONNECTION_TIMED_OUT
провайдер молодец... есть какой-либо выход частный для кинопаб? (не хотелось бы ставить VPN на роутер для всего трафика)

через VPN на ПК результат
404 Not Found
nginx

**Reply by stavultras** [reply]
А это уже вопрос к самому Kinopub и его владельцам/разработчикам. Тут только аддон обсуждают.

**Reply by stavultras** [reply]
Оно конечно можно извратиться, поставить в libreelec openvpn, в нём подкинуть профиль самого VPN (купленного/бесплатного/другое), отключить назначение дефолтного шлюза и прокинуть через VPN только айпишник самого Kinopub API (через таблицу маршрутизации например или iptables). Видео траффик через API не гоняется, через него только метаданные приходят, минимальный траффик в общем.

**Reply by ktchatskiy** [reply]
Спасибо за помощь!

**Reply by ktchatskiy** [reply]
> Оно конечно можно извратиться, поставить в libreelec openvpn, в нём подкинуть профиль самого VPN (купленного/бесплатного/другое), отключить назначение дефолтного шлюза и прокинуть через VPN только айпишник самого Kinopub API (через таблицу маршрутизации например или iptables). Сам траффик через API не гоняется, через него только метаданные приходят, минимальный траффик в общем.

наверное я не совсем понимаю, но
https://api.service-kp.com/v1 если отредактировать на https://api.service-kp.com, то открывается и резолвится на https://kinoapi.com/ без VPN нормально. Теперь не уверен, что дело в провайдере или я перегрелся? )

**Reply by stavultras** [reply]
хммм. не совсем понимаю как https://api.service-kp.com/v1 не работает, а https://api.service-kp.com/ - работает )) один и тот же IP.....

**Reply by ktchatskiy** [reply]
> хммм. не совсем понимаю как https://api.service-kp.com/v1 не работает, а https://api.service-kp.com/ - работает )) один и тот же IP.....

вот и я не понимаю...
